### PR TITLE
object_sa OK

### PR DESCRIPTION
**Pull request by louist103**
There is one block of `unaccounted` that I cant figure out but other than that everything is named.

### CODE REVIEW

File: undefined_syms.txt [hunk 1]
```diff
@@ -2194,12 +2179,6 @@ D_06001C80 = 0x06001C80;
 // z_en_sa
 D_02005730 = 0x02005730;
 D_02010E20 = 0x02010E20;
```
Comment (AngheloAlf) [inline undefined_syms.txt:2181]: I know is a bit out of scope, but maybe you could take a look at those cutscenes and name them too. This way Saria would be complete :D

File: assets/xml/objects/object_sa.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        <Texture Name="gSariaEyeTULT" OutName="eye_tult" Format="rgb5a1" Width="18" Height="6" Offset="0x2B00"/>
+        <Texture Name="gSariaSkinTULT" OutName="skin_tult" Format="rgb5a1" Width="8" Height="9" Offset="0x2A70"/>
+        <Texture Name="gSariaUnkTULT" OutName="unk_tult" Format="rgb5a1" Width="16" Height="9" Offset="0x2BD8"/>
+
+        <Texture Name="gSariaBootsTex" OutName="boots" Format="ci8" Width="16" Height="16" Offset="0x23F0"/>
+        <Texture Name="gSariaGreenTex" OutName="green" Format="ci8" Width="8" Height="8" Offset="0x24F0"/> 
+        <Texture Name="gSariaKneeTex" OutName="knee" Format="ci8" Width="16" Height="16" Offset="0x2570"/>
+        <Texture Name="gSariaTorsoTex" OutName="torso" Format="ci8" Width="16" Height="16" Offset="0x2670"/>
+        <Texture Name="gSariaBeltTex" OutName="belt" Format="ci8" Width="16" Height="16" Offset="0x2770"/>
+        <Texture Name="gSariaUnkGreenTex" OutName="IDK" Format="ci8" Width="16" Height="8" Offset="0x2870"/>
```
Comment (AngheloAlf) [inline assets/xml/objects/object_sa.xml:81]: This OutName doesn't seem appropriate.

File: assets/xml/objects/object_sa.xml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        <Texture Name="gSariaUnusedHandTex" OutName="unused_hand" Format="ci8" Width="16" Height="16" Offset="0x28F0"/>
+        <Texture Name="gSariaSleeveTex" OutName="sleeve" Format="ci8" Width="8" Height="16" Offset="0x29F0"/>
+        <Texture Name="gSariaHairTex" OutName="hair" Format="ci8" Width="8" Height="16" Offset="0x2EC8"/>
+        <Texture Name="gSariaEyeOpenTex" OutName="eye_open" Format="ci8" Width="32" Height="32" Offset="0x2F48"/>
+        <Texture Name="gSariaMouthClosedTex" OutName="mouth_closed" Format="ci8" Width="32" Height="16" Offset="0x3348"/>
+        <Texture Name="gSariaSkinTex" OutName="skin" Format="ci8" Width="8" Height="8" Offset="0x3548"/>
+        <Texture Name="gSariaMouthClosed2Tex" OutName="mouth_closed2" Format="ci8" Width="32" Height="16" Offset="0x3588"/>
+        <Texture Name="gSariaEarTex" OutName="ear" Format="ci8" Width="8" Height="8" Offset="0x3788"/>
+        <Texture Name="gSariaCollarTex" OutName="collar" Format="ci8" Width="8" Height="8" Offset="0x37C8"/>
+        <Texture Name="gSariaHandTex" OutName="Hand" Format="ci8" Width="8" Height="8" Offset="0x3808"/>
```
Comment (AngheloAlf) [inline assets/xml/objects/object_sa.xml:91]: lowercase for OutNames
```suggestion
        <Texture Name="gSariaHandTex" OutName="hand" Format="ci8" Width="8" Height="8" Offset="0x3808"/>
```

**Review by AngheloAlf — COMMENTED** [review]
About the unaccounted you have, it is between other textures, so I guess it should be a texture...
You already checked if is there any other actor that uses `object_sa`, right?

File: assets/xml/objects/object_sa.xml (hunk 2 above)
Comment (louist103) [reply assets/xml/objects/object_sa.xml:81]: that wasn't supposed to be left in lol. Ill take it out in the next PR.

File: src/overlays/actors/ovl_Demo_Sa/z_demo_sa.c [hunk 4]
```diff
@@ -40,16 +41,33 @@ void func_8098FC9C(DemoSa* this, GlobalContext* globalCtx);
...
+void DemoSa_DrawNothing(DemoSa* this, GlobalContext* globalCtx);
+void DemoSa_DrawOpa(DemoSa* this, GlobalContext* globalCtx);
+void DemoSa_DrawXlu(DemoSa* this, GlobalContext* globalCtx);
+
+typedef enum {
+    /* 0 */ SARIA_EYE_OPEN,
+    /* 1 */ SARIA_EYE_HALF,
+    /* 2 */ SARIA_EYE_CLOSED,
+    /* 3 */ SARIA_EYE_SUPRISED,
+    /* 4 */ SARIA_EYE_SAD,
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Demo_Sa/z_demo_sa.c:53]: Trailing comma is giving a warning, please remove it.
```suggestion
    /* 4 */ SARIA_EYE_SAD
```

File: src/overlays/actors/ovl_Demo_Sa/z_demo_sa.c [hunk 5]
```diff
@@ -40,16 +41,33 @@ void func_8098FC9C(DemoSa* this, GlobalContext* globalCtx);
...
+} SariaEyeState;
+
+typedef enum {
+    /* 0 */ SARIA_MOUTH_CLOSED2,
+    /* 1 */ SARIA_MOUTH_SUPRISED,
+    /* 2 */ SARIA_MOUTH_CLOSED,
+    /* 3 */ SARIA_MOUTH_SMILING_OPEN,
+    /* 4 */ SARIA_MOUTH_FROWNING,
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Demo_Sa/z_demo_sa.c:61]:
```suggestion
    /* 4 */ SARIA_MOUTH_FROWNING
```

File: src/overlays/actors/ovl_En_Sa/z_en_sa.c [hunk 6]
```diff
@@ -16,6 +17,22 @@ void func_80AF683C(EnSa* this, GlobalContext* globalCtx);
 void func_80AF68E4(EnSa* this, GlobalContext* globalCtx);
 void func_80AF6B20(EnSa* this, GlobalContext* globalCtx);
 
+typedef enum {
+    /* 0 */ SARIA_EYE_OPEN,
+    /* 1 */ SARIA_EYE_HALF,
+    /* 2 */ SARIA_EYE_CLOSED,
+    /* 3 */ SARIA_EYE_SUPRISED,
+    /* 4 */ SARIA_EYE_SAD,
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Sa/z_en_sa.c:25]:
```suggestion
    /* 4 */ SARIA_EYE_SAD
```

File: src/overlays/actors/ovl_En_Sa/z_en_sa.c [hunk 7]
```diff
@@ -16,6 +17,22 @@ void func_80AF683C(EnSa* this, GlobalContext* globalCtx);
...
+} SariaEyeState;
+
+typedef enum {
+    /* 0 */ SARIA_MOUTH_CLOSED2,
+    /* 1 */ SARIA_MOUTH_SUPRISED,
+    /* 2 */ SARIA_MOUTH_CLOSED,
+    /* 3 */ SARIA_MOUTH_SMILING_OPEN,
+    /* 4 */ SARIA_MOUTH_FROWNING,
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Sa/z_en_sa.c:33]:
```suggestion
    /* 4 */ SARIA_MOUTH_FROWNING
```

**Review by AngheloAlf — CHANGES_REQUESTED** [review]

**Review by fig02 — APPROVED** [review]

File: assets/xml/objects/object_sa.xml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        <DList Name="gSariaRightFootDL" Offset="0xA810"/>
+        <DList Name="gSariaLeftThighDL" Offset="0xA9D0"/>
+        <DList Name="gSariaLeftShinDL" Offset="0xAC20"/>
+        <DList Name="gSariaLeftFootDL" Offset="0xAEE0"/>
+
+        <Texture Name="gSariaClothesTULT" OutName="clothes_tult" Format="rgb5a1" Width="18" Height="6" Offset="0x21F0"/>
+        <Texture Name="gSariaMouthTULT" OutName="mouth_tult" Format="rgb5a1" Width="18" Height="6" Offset="0x2CF8"/>
+        <Texture Name="gSariaEyeTULT" OutName="eye_tult" Format="rgb5a1" Width="18" Height="6" Offset="0x2B00"/>
+        <Texture Name="gSariaSkinTULT" OutName="skin_tult" Format="rgb5a1" Width="8" Height="9" Offset="0x2A70"/>
+        <Texture Name="gSariaUnkTULT" OutName="unk_tult" Format="rgb5a1" Width="16" Height="9" Offset="0x2BD8"/>
```
Comment (Roman971) [inline assets/xml/objects/object_sa.xml:73]: It's actually "TLUT". Relevant for the other ones above
```suggestion
        <Texture Name="gSariaUnkTLUT" OutName="unk_tlut" Format="rgb5a1" Width="16" Height="9" Offset="0x2BD8"/>
```

File: assets/xml/objects/object_sa.xml [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        <Limb Name="gSariaRightHandLimb" LimbType="Standard" Offset="0xB148"/>
+        <Limb Name="gSariaHeadLimb" LimbType="Standard" Offset="0xB154"/>
+
+        <Animation Name="gSariaSealGanonAnim" Offset="0x1334"/>
+        <Animation Name="gSariaOcarinaToMouthAnim" Offset="0x1D50"/>
+        <Animation Name="gSariaWaitArmsToSideAnim" Offset="0x21D8"/>
+        <Animation Name="gSariaPlayingOcarinaAnim" Offset="0xBAEC"/>
+        <Animation Name="gSariaStopPlayingOcarinaAnim" Offset="0xC500"/>
+        <Animation Name="gSariaLinkLearnedSariasSongAnim" Offset="0xCFD8"/>
+        <Animation Name="gSariaReturnToOcarina" Offset="0xD7E8"/>
```
Comment (Roman971) [inline assets/xml/objects/object_sa.xml:28]: A few of these animations seem to be missing the `Anim` suffix

**Review by fig02 — APPROVED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**